pkg/odo/cli/delete/namespace/namespace_test.go

```diff
@@ -0,0 +1,81 @@
+package namespace
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/redhat-developer/odo/pkg/kclient"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	_delete "github.com/redhat-developer/odo/pkg/project"
+)
+
+func TestCmdNamespaceDelete(t *testing.T) {
+	type fields struct {
+		commandName           string
+		namespaceName         string
+		forceFlag             bool
+		kubernetesClient      func(ctrl *gomock.Controller) kclient.ClientInterface
+		deleteNamespaceClient func(ctrl *gomock.Controller) _delete.Client
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name: "No namespace found",
+			fields: fields{
+				commandName:   "namespace",
+				namespaceName: "my-namespace",
+				forceFlag:     false,
+				kubernetesClient: func(ctrl *gomock.Controller) kclient.ClientInterface {
+					client := kclient.NewMockClientInterface(ctrl)
+					return client
+				},
+				deleteNamespaceClient: func(ctrl *gomock.Controller) _delete.Client {
+					client := _delete.NewMockClient(ctrl)
+					client.EXPECT().Exists("my-namespace").Return(false, nil)
+					return client
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Delete namespace",
+			fields: fields{
+				commandName:   "namespace",
+				namespaceName: "my-namespace",
+				forceFlag:     false,
```

Comment:
```suggestion
				forceFlag:     true,
```

I think you need to set `forceFlag` to `true` if you want to test the behavior when the namespace exists.
And in this case, you can also check that `client.Delete` has been called exactly once.